BrowserKit/Sources/ToolbarKit/AddressToolbar/AddressToolbarDelegate.swift

```diff
@@ -10,4 +10,5 @@ public protocol AddressToolbarDelegate: AnyObject {
     func openBrowser(searchTerm: String)
     func openSuggestions(searchTerm: String)
     func addressToolbarAccessibilityActions() -> [UIAccessibilityCustomAction]?
+    func configureCFR(_ addressToolbar: BrowserAddressToolbar, for button: ToolbarButton)
```

Comment:
to better understand this PR, one question - I'm not really familiar with using the delegate pattern with Redux, could you explain why we need to create a delegate or the usage of it compared to dispatching an action to configure the CFR and updating the view state to whether the CFR is needed?

Reply:
The hurdle I ran into when trying to do this through redux was that the CFR needs a UI component to anchor to, and it didn't seem like common practice to bloat the states with UI objects... though that was just an assumption. Do you think that is better handled via redux?

Reply:
Spoke with @thatswinnie and we decided to investigate all delegate usage by the new toolbars at once in phase 2. #21638 was created to track this